src/shared/elf-util.c

```diff
@@ -614,6 +620,11 @@ static int parse_core(int fd, const char *executable, char **ret, sd_json_varian
         if (!c.dwfl)
                 return log_warning_errno(SYNTHETIC_ERRNO(EINVAL), "Could not parse core file, dwfl_begin() failed: %s", sym_dwfl_errmsg(sym_dwfl_errno()));
 
+#if HAVE_DWFL_SET_SYSROOT
+        if (root && sym_dwfl_set_sysroot(c.dwfl, root) < 0)
```

Comment:
This could fail because it's enabled at build time, but not available at runtime - it's dlopen so we are not generating a hard dependency. So check for that case, and log but continue if it happens

Reply:
How do I check for that, `sym_dwfl_set_sysroot == NULL` ?

Reply:
yeah checking for that before root should do the trick